src/main/java/de/jamba/hudson/plugin/wsclean/PrePostClean.java

```diff
@@ -88,42 +94,62 @@ private void executeOnSlaves(AbstractBuild build, BuildListener listener) {
 			for (Node node : nodesForLabel) {
 				if (!runNode.equals(node.getNodeName())) {
 					String normalizedName = "".equals(node.getNodeName()) ? "master" : node.getNodeName();
-						listener.getLogger().println(
-								"cleaning on " + normalizedName);
-						deleteWorkspaceOn(project, listener, node, normalizedName);
+					listener.getLogger().println("cleaning on " + normalizedName);
+					deleteWorkspaceOn(project, listener, node, normalizedName);
 				}
-
 			}
 		}
 	}
 
 	@SuppressWarnings("rawtypes")
 	private void deleteWorkspaceOn(AbstractProject project, BuildListener listener, Node node, String nodeName) {
+		FilePath fp = null;
 		if (project instanceof TopLevelItem) {
-			FilePath fp = node.getWorkspaceFor((TopLevelItem) project);
-			if (fp != null) {
-				try {
-					fp.deleteContents();
-				} catch (IOException e) {
-					listener.getLogger().println(
-							"can't delete on node " + nodeName + "\n" + e.getMessage());
-					listener.getLogger().print(e);
-				} catch (InterruptedException e) {
-					listener.getLogger().println(
-							"can't delete on node " + nodeName + "\n" + e.getMessage());
-					listener.getLogger().print(e);
-				} catch (RequestAbortedException e){
-					listener.getLogger().println(
-							"can't delete on node " + nodeName + "\n" + e.getMessage());
-				}
-				
-			} else {
+			fp = node.getWorkspaceFor((TopLevelItem) project);
+		}
+		else if (project instanceof MatrixConfiguration) {
+			fp = getWorkspaceFor((MatrixConfiguration) project, listener, node, nodeName);
```

Comment:
Matrix builds have multiple workspaces - there's the workspace where the "main" build runs, plus another for every sub-build that the matrix used.
Any support for matrix builds needs to handle "N workspaces" rather than assuming "1 workspace".